src/tools/clippy/Cargo.toml

```diff
@@ -39,6 +39,7 @@ filetime = "0.2"
 rustc-workspace-hack = "1.0"
 
 # UI test dependencies
+ui-test-dependencies = { path = "ui-test-dependencies" }
 clippy_utils = { path = "clippy_utils" }
```

Comment:
Does Clippy still need all of those dependencies in the top level Cargo.toml or is it enough to have them all in ``ui-test-dependencies``?

Reply:
Good callout :) I think it's enough to only have them in ui-test-dependencies? but there's a comment in compiletest that makes me wary:
```
// Test dependencies may need an `extern crate` here to ensure that they show up
// in the depinfo file (otherwise cargo thinks they are unused)
#[allow(unused_extern_crates)]
extern crate derive_new;
```